WcaOnRails/app/views/competitions/_my_competitions_table.html.erb

```diff
@@ -0,0 +1,54 @@
+<% if competitions.length < 1 %>
+  <% message = upcoming ? "You currently have no upcoming competitions! Check out the #{link_to 'competitions list', '/results/competitions.php'}.".html_safe : "You have no past competitions." %>
+  <div class="alert alert-info" role="alert">
+    <%= message %>
+  </div>
+<% else %>
+  <table class="table competitions-table floatThread <%= !upcoming ? "table-striped" : "" %>">
+    <tr>
+      <th>Name</th>
+      <th>Location</th>
+      <th>Date</th>
+      <th></th>
+      <th></th>
+      <th class="big-column"></th>
+    </tr>
+    <% competitions.each do |competition| %>
+      <% if competition.isConfirmed %>
+        <% if competition.showAtAll %>
+          <% message = "This competition is confirmed and visible." %>
+        <% else %>
+          <% message = "This competition is confirmed but not visible." %>
+        <% end %>
+      <% else %>
+        <% if competition.showAtAll %>
+          <% message = "This competition is not confirmed but is visible." %>
+        <% else %>
+          <% message = "This competition is not confirmed and not visible." %>
+        <% end %>
+      <% end %>
+      <% if upcoming %>
+        <tr class="<%= competition.isConfirmed ? "confirmed" : "not-confirmed" %> <%= competition.showAtAll ? "visible" : "not-visible" %>" data-toggle="tooltip" data-placement="top" data-container="body" title="<%= message %>">
+      <% else %>
+        <tr>
```

Comment:
Hmmm... this is quite weird. I don't like how the `if` statement ends inside of the `<tr>`. It's particularly weird that the indentation of the `<tds>` is the same level as the `<tr>`. I think this would be cleaner if we added another class for `past`/`not-past`. I think that `upcoming` is a bad term, because we treat ongiong competitions as upcoming, right?

To disable the tooltip for past competitions, we should just set message to empty string if the competition is past.